### PR TITLE
feat(dev-tools): update actions to display function name instead of callFrom

### PR DESCRIPTION
**Pull request by y77cao**
Follow up for https://github.com/latticexyz/mud/pull/1466

Before:
Screenshot 2023-09-14 at 9 45 53 PM
After
Screenshot 2023-09-14 at 9 42 26 PM

### CODE REVIEW

File: packages/dev-tools/src/actions/WriteSummary.tsx [hunk 1]
```diff
@@ -56,6 +56,11 @@ export function WriteSummary({ write }: Props) {
           .filter(isDefined)
       : null;
 
+  const functionSelectorAndArgs: Hex = write.request?.args?.length
```
Comment (holic) [inline packages/dev-tools/src/actions/WriteSummary.tsx:59]: I think this won't always be the case if a system is called with arguments. Unfortunately the template doesn't have a good example of this but I think the example/minimal does (look for `InventorySystem.sol`).

Instead we might have to check if the request method matches `call` or `callFrom` and then decode accordingly.

File: packages/dev-tools/src/actions/WriteSummary.tsx [hunk 2]
```diff
@@ -56,6 +56,17 @@ export function WriteSummary({ write }: Props) {
           .filter(isDefined)
       : null;
 
+  let functionName = write.request.functionName;
+  let functionArgs = write.request.args;
+  if (functionName === "call" || functionName === "callFrom") {
```
Comment (holic) [inline packages/dev-tools/src/actions/WriteSummary.tsx:61]: note that `call` and `callFrom` have different signatures:
https://github.com/latticexyz/mud/blob/4c7fd3eb29e3d3954f2f1f36ace474a436082651/packages/world/src/interfaces/IWorldKernel.sol#L16-L32

Comment (y77cao) [reply packages/dev-tools/src/actions/WriteSummary.tsx:61]: I see. Looks like funcSelectorAndArgs is the last param for both so should be ok here?

Comment (holic) [reply packages/dev-tools/src/actions/WriteSummary.tsx:61]: oh yep, you're right, I missed that you were grabbing the last element, not slicing off the first element 🙈 my bad

File: packages/dev-tools/src/actions/WriteSummary.tsx [hunk 3]
```diff
@@ -73,7 +84,7 @@ export function WriteSummary({ write }: Props) {
         )}
       >
         <div className="flex-1 font-mono text-white whitespace-nowrap overflow-hidden text-ellipsis">
-          {write.request.functionName}({write.request.args?.map((value) => serialize(value)).join(", ")})
+          {functionName}({functionArgs?.map((value) => serialize(value)).join(", ")})
```
Comment (holic) [inline packages/dev-tools/src/actions/WriteSummary.tsx:87]:
```suggestion
          {functionName}({functionArgs?.map((value) => serialize(value)).join(", ")})
          {" "}
          {write.request.functionName !== functionName ? <span className="text-xs text-white/40">via {write.request.functionName}</span> : null}
```

**Review by holic — APPROVED** [review]
Looks great!

My comment above isn't blocking. If you do decide to commit that, it might need a pass through prettier because I just typed it into GitHub and no idea if I got the formatting correct.

**Comment by holic** [comment]
just realized we can actually land this independent of #1466!

**Review by holic — APPROVED** [review]

**Pull request merged**